projectq/ops/_gates.py

```diff
@@ -74,17 +77,29 @@ def matrix(self):
 
 class XGate(SelfInverseGate):
     """ Pauli-X gate class """
+
     def __str__(self):
         return "X"
 
     @property
     def matrix(self):
         return np.matrix([[0, 1], [1, 0]])
 
+    def get_commutable_circuit_list(self, n=0):
+        """ Sets _commutable_circuit_list for C(NOT, n) where
+        n is the number of controls """
+        if (n == 1):
+            # i.e. this is a CNOT gate (one control)
+            # We define the qubit with the NOT as qb0, the qubit with 
+            # the control as qb1, then all next qbs are labelled 2,3 etc.
+            commutable_circuit_list = [[RelativeCommand(H,(0,)), RelativeCommand(C(NOT),(2,), relative_ctrl_idcs=(0,)), RelativeCommand(H,(0,))]]
```

Comment:
Are you sure about this circuit equivalence?

Reply:
I got this commutation rule from this paper:
https://arxiv.org/abs/1710.07345#:~:text=Automated%20optimization%20of%20large%20quantum%20circuits%20with%20continuous%20parameters,-Yunseong%20Nam%2C%20Neil&text=In%20particular%2C%20we%20provide%20better,of%20the%20underlying%20quantum%20algorithms.
(Figure 5)